### ISSUE TITLE
Is it possible to use "//go:embed icon.png" to create a QIcon?

### ISSUE THREAD

**Issue by mark-summerfield** [mappu/miqt#63]
Is it possible to use "//go:embed icon.png" to create a QIcon?
I tried creating a QIconEngine but it needs a pointer & I can't work out how to do it. I have:
```
//go:embed icon.png
var Icon []byte
```
and want to create a function:
```
func getIcon(name string) *qt.QIcon { ... }
```

**Comment by jamalsa** [comment]
You can use [rcc](https://doc.qt.io/qt-6/rcc.html) from Qt for this. For example if you have following folders:

```
icons\
  icon.png
icons.qrc
```

`icons.qrc` content:

```
<!DOCTYPE RCC>
<RCC version="1.0">
<qresource>
    <file>icons/icon.png</file>
</qresource>
</RCC>
```

Then run `rcc --binary -o icons.rcc icons.qrc` to generate rcc file. You can use the result using go embed:

```go
package main

import (
        "embed"

       "github.com/mappu/miqt"
)

//go:embed icons.rcc
var Icons []byte

func main() {
   // Load RCC
   qt.QResource_RegisterResourceWithRccData(&Icons[0])
  ...
  // Use it
  icon :=  qt.NewQIcon4(":/icons/icon.png")

}
```

**Comment by mark-summerfield** [comment]
Yes, I'd seen an example like that. But was hoping to do it the more Go-ish way.

**Comment by mappu** [comment]
The rcc way is nice when using Qt Designer and having a lot of image assets.

A shorter way would be to use one of the Qt functions that loads an image from binary data. Something like QPixmap.loadFromData() or NewQImage() both support loading data from a []byte.

**Comment by mark-summerfield** [comment]
I never use Qt Designer; I prefer to do all my widget creation & layouts in code. Good points you make about NewQImage(); I'll try it.
Thanks.

**Comment by mark-summerfield** [comment]
I just tried
```
//go:embed go-up.png
var Icon []byte
...
	icon := QImage_FromDataWithData(Icon)
	app.window.SetWindowIcon(icon)
```
But get this error:
```
undefined: QImage_FromDataWithData
```
My `go.mod` contains:
```
require github.com/mappu/miqt v0.6.0
```
and my import is
```
import 	qt "github.com/mappu/miqt/qt6"
```
So I'm not sure what I'm doing wrong.

**Comment by mappu** [comment]
If that code snippet is correct, I think you are missing the package prefix on the function call?

```diff
- icon := QImage_FromDataWithData(Icon)
+ icon := qt.QImage_FromDataWithData(Icon) 
```

**Comment by mark-summerfield** [comment]
I'm very sorry, I forgot to do that!

This now works correctly:
```
//go:embed icon.png
var Icon []byte
...
	app.window = qt.NewQMainWindow3(nil, qt.Dialog)
	app.window.SetWindowIcon(getIcon(Icon))
...
func getIcon(data []byte) *qt.QIcon {
	image := qt.QImage_FromDataWithData(data)
	pixmap := qt.QPixmap_FromImage(image)
	return qt.NewQIcon2(pixmap)
}
```
However, if I replace the icon with `icon.svg`, it silently fails.

**Comment by mappu** [comment]
OK, it sounds like everything in this issue is resolved, the remaining work is tracked in #62.
